handler/user_handler.go

```diff
@@ -60,3 +60,45 @@ func LoginHandler(userService service.UserService) http.HandlerFunc {
 		w.Write(responseBytes)
 	}
 }
+
+func ListUsersHandler(userService service.UserService) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		w.Header().Set("Content-Type", "application/json")
+
+		if r.Method == "GET" {
+			user, err := userService.ListUsers(r.Context())
+
+			if err == service.NoUsersExist {
+				fmt.Println("handler: No users exist")
+
+				w.WriteHeader(http.StatusNoContent)
+				responseBytes, _ := json.Marshal(contract.ErrorResponse{Error: "no user found"})
+				w.Write(responseBytes)
+				return
+
+			}
+
+			if err != nil {
+				fmt.Printf("handler: error while searching for user,error= %s", err.Error())
+
+				w.WriteHeader(http.StatusInternalServerError)
+				responseBytes, _ := json.Marshal(contract.ErrorResponse{Error: "something went wrong"})
+				w.Write(responseBytes)
+				return
+			}
+			//write a loop to convert domain object to contract object
+
+			var userResp = make([]contract.User, len(user))
```

Comment:
here u can just initialize slice with make([]contract.User, 0) instead of len(user)